### PR TITLE
feat(auth): Added fetchOptions configuration on auth module

### PR DESCRIPTION
**Pull request by dgonsan**
Issue #, if available:
#6905

Description of changes:

Added fetchOptions to Auth module configuration, it's a new  optional parameter that allows to use the already existing fetchOptions configuration available in amazon-cognito-idenity-js from the auth module  and its components.

The changes are just a simple pass through of the new optional parameter from the Auth class to the CognitoUserPool class that makes use of it.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/index.d.ts [hunk 1]
```diff
@@ -308,6 +308,7 @@ declare module 'amazon-cognito-identity-js' {
 		UserPoolId: string;
 		ClientId: string;
 		endpoint?: string;
+        fetchOptions?: any;
```
Comment (drochetti) [inline packages/amazon-cognito-identity-js/index.d.ts:311]: I would like to see an actual type. What options are available?

nit: fix formatting

File: packages/auth/src/types/Auth.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export interface AuthOptions {
 	identityPoolRegion?: string;
 	clientMetadata?: any;
 	endpoint?: string;
+	fetchOptions?: any;
```
Comment (drochetti) [inline packages/auth/src/types/Auth.ts:54]: use the type defined in `packages/amazon-cognito-identity-js/index.d.ts`

**Review by drochetti — CHANGES_REQUESTED** [review]

File: packages/amazon-cognito-identity-js/index.d.ts (hunk 1 above)
Comment (dgonsan) [reply packages/amazon-cognito-identity-js/index.d.ts]: I added a IFetchOptions type to amazon-cognito-idenity-js with the proper typing and restrictions for fetch options.

File: packages/auth/src/types/Auth.ts (hunk 2 above)
Comment (dgonsan) [reply packages/auth/src/types/Auth.ts]: I didn't find any define type for fetchOptiions there so I added IFetchOptions one with the fetch options definition.

**Review by evcodes — COMMENTED** [review]
This looks good to me, just need to fix the merge conflicts so we can merge in your first PR 💯  

Thank you for contributing to Amplify 👍🏼

**Comment by dgonsan** [comment]
I rebased all the changes on the main branch and solved the conflicts.

**Comment by dgonsan** [comment]
Hey @evcodes , sorry for the delay I was busy with other projects and didn't have time to come back to this pull request until now.

1. What tests do you want me to add? Because there is no new feature here, the feature is in the cognito module and was implemented long time ago, this just adds the option to pass the values to the other module to use that feature.

2. It's just and additional configuration option, probably should be added in this section of the documentation: https://docs.amplify.aws/lib/auth/start/q/platform/js/#configure-your-application

```
    // OPTIONAL: Configuration for fetch options on authentication requests
    fetchOptions: {
        // OPTIONAL - Authentication credentials mode
        // See: https://github.github.io/fetch/#credentials
        credentials: 'same-origin' | 'include' | 'omit';
    }
```

This allows retrieving/sending credentials in the authentication request which in our case is used to set additional cookies required by our application.

**Comment by cwomack** [comment]
Hey, @dgonsan sorry for the delays in communication on this PR.  With the release of Amplify v6, the `amazon-cognito-identity-js package` is no longer used (was last used in v5.x).  Would you be able to see if the new `fetchAuthSession` API in v6 is sufficient to accomplish what you're looking for in this PR ([link](https://docs.amplify.aws/react/build-a-backend/auth/manage-user-session/#retrieve-a-user-session))? If not, can you elaborate a little more on what the use case is that you're looking for? Thank you!

**Comment by cwomack** [comment]
Closing this issue as we have not heard back from you. If you are still experiencing this, please feel free to reply back and provide any information previously requested and we'd be happy to re-open the issue.

Thank you!

**Pull request closed without merging**